cmd/repo-updater/repos/store_test.go

```diff
@@ -855,6 +855,120 @@ func testStoreSetClonedRepos(store repos.Store) func(*testing.T) {
 	}
 }
 
+func testStoreCountClonedRepos(store repos.Store) func(*testing.T) {
+	clock := repos.NewFakeClock(time.Now(), 0)
+	now := clock.Now()
+
+	return func(t *testing.T) {
+		t.Helper()
+
+		github := repos.Repo{
+			Name:        "github.com/foo/bar",
+			URI:         "github.com/foo/bar",
+			Description: "The description",
+			Language:    "barlang",
+			CreatedAt:   now,
+			Cloned:      false,
+			ExternalRepo: api.ExternalRepoSpec{
+				ID:          "AAAAA==",
+				ServiceType: "github",
+				ServiceID:   "http://github.com",
+			},
+			Sources: map[string]*repos.SourceInfo{
+				"extsvc:1": {
+					ID:       "extsvc:1",
+					CloneURL: "git@github.com:foo/bar.git",
+				},
+			},
+			Metadata: new(github.Repository),
+		}
+
+		gitlab := repos.Repo{
+			Name:        "gitlab.com/foo/bar",
+			URI:         "gitlab.com/foo/bar",
+			Description: "The description",
+			Language:    "barlang",
+			CreatedAt:   now,
+			Cloned:      false,
+			ExternalRepo: api.ExternalRepoSpec{
+				ID:          "1234",
+				ServiceType: extsvc.TypeGitLab,
+				ServiceID:   "http://gitlab.com",
+			},
+			Sources: map[string]*repos.SourceInfo{
+				"extsvc:2": {
+					ID:       "extsvc:2",
+					CloneURL: "git@gitlab.com:foo/bar.git",
+				},
+			},
+			Metadata: new(gitlab.Project),
+		}
+
+		bitbucketServer := repos.Repo{
+			Name:        "bitbucketserver.mycorp.com/foo/bar",
+			URI:         "bitbucketserver.mycorp.com/foo/bar",
+			Description: "The description",
+			Language:    "barlang",
+			CreatedAt:   now,
+			Cloned:      false,
+			ExternalRepo: api.ExternalRepoSpec{
+				ID:          "1234",
+				ServiceType: "bitbucketServer",
+				ServiceID:   "http://bitbucketserver.mycorp.com",
+			},
+			Sources: map[string]*repos.SourceInfo{
+				"extsvc:3": {
+					ID:       "extsvc:3",
+					CloneURL: "git@bitbucketserver.mycorp.com:foo/bar.git",
+				},
+			},
+			Metadata: new(bitbucketserver.Repo),
+		}
+
+		repositories := repos.Repos{
+			&github,
+			&gitlab,
+			&bitbucketServer,
+		}
```

Comment:
That's a lot of lines for a test like this 😄 All you need is different repositories with different `Cloned` attributes, right? Can't you do something like ...
```suggestion
		var repositories repos.Repos
	for i := 0; i < 3; i++ {
		repositories = append(repositories, repos.Repo{
 			Name:        fmt.Sprintf("github.com/%d/%d",i,i),
 			URI:         fmt.Sprintf("github.com/%d/%d",i,i),
 			Cloned:      false,
 			ExternalRepo: api.ExternalRepoSpec{
 				ID:          fmt.Sprintf("%d", i),
 				ServiceType: extsvc.TypeGitHub,
 				ServiceID:   "http://github.com",
 			},
 			Sources: map[string]*repos.SourceInfo{
 				"extsvc:3": {
 					ID:       "extsvc:3",
 					CloneURL: "git@github.com:foo/bar.git",
 				},
 			},
 			Metadata: new(github.Repo),
 		}
	}
```

Reply:
Indeed, i copy pasted the code of the test above it and tried to adapt until it worked. I definitely need to simplify it 👍